pkg/odo/cli/component/create.go

```diff
@@ -942,109 +936,137 @@ func (co *CreateOptions) downloadStarterProject(projectPassed string, interactiv
 		return err
 	}
 
-	var logUrl, url, sparseDir string
-	if project.Git != nil {
-		if strings.Contains(project.Git.Location, "github.com") {
-			url, err = util.GetGitHubZipURL(project.Git.Location, project.Git.Branch, project.Git.StartPoint)
-			if err != nil {
-				return err
-			}
-			sparseDir = project.Git.SparseCheckoutDir
```

Comment:
are we not supporting `sparseCheckoutDir` for git like projects?

Reply:
we should not. This is quite hard to do in git. We can look into it later

Reply:
the openliberty devfile is using it https://github.com/odo-devfiles/registry/blob/master/devfiles/java-openliberty/devfile.yaml#L10 :), need to update this.